observability/WFLY-14854_add_opentelemetry_support.adoc

```diff
@@ -0,0 +1,76 @@
+ = Add OpenTelemetry Tracing Support to WildFly
+:author:            Jason Lee
+:email:             jasondlee@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+https://opentelemetry.io[OpenTelemetry] is an ambitious specification that aims to encapsulate APIs for tracing, metrics, and logging. With OpenTracing (which WildFly already supports) having shutdown in favor of OpenTelemetry, we need to migrate our support as well. This project will only cover the tracing aspects of OpenTelemetry. Any future support for the metrics and logging parts of the OpenTelemetry spec will need to be handled by a separate effort.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14854
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1688
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+* [X] Engineering
+* [X] QE
+
+=== Affected Projects or Components
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+* [ ] Managed domain
+* [x] OpenShift s2i
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* The server/domain administrator should be able to configure OpenTelemetry export and span processor options
+* Jakarta REST endpoints should be automatically traced
+* Adherence to the Observability Standard for Runtimes
+* Context propagation should work for both incoming and outgoing request (e.g., a `ClientRequestFilter`)
+
+=== Nice-to-Have Requirements
+
+* Feature parity with current OpenTracing support, if/where it makes sense.
+* Tracing of JDBC calls
+* Tracing of EJB calls
```

Comment:
Would they be traced by default (similarly as the Jakarta REST endpoints)?

Reply:
I would imagine so, but there's been little discussion on the topic to date.

Reply:
Is it something opentracing provide today?

Reply:
It is not something we offer. There are ways to do it, and we have open RFEs to implement that, but we do not currently support that, so it's not a loss of functionality.